pinot-broker/src/main/java/org/apache/pinot/broker/routing/segmentpartition/SegmentPartitionMetadataManager.java

```diff
@@ -136,7 +138,68 @@ private static List<String> getOnlineServers(ExternalView externalView, String s
     return onlineServers;
   }
 
-  private void computeTablePartitionInfo() {
+  @Override
+  public synchronized void onAssignmentChange(IdealState idealState, ExternalView externalView,
```

Comment:
note: GH is not great at showing this but I have simply moved the compute table partition info methods towards the end. however reviewers can double check this claim.